db/migrate/20240820192904_fix_used_by_grader_and_visible_for_task_files.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class FixUsedByGraderAndVisibleForTaskFiles < ActiveRecord::Migration[7.1]
+  class TaskFile < ApplicationRecord
+  end
+
+  def change
+    visible_options = %w[yes no delayed]
+    TaskFile.where.not(visible: visible_options).or(TaskFile.where(used_by_grader: nil)).find_each do |task_file|
+      task_file.visible = {'Task' => 'yes', 'Test' => 'no', 'ModelSolution' => 'delayed'}[task_file.fileable_type] unless visible_options.include? task_file.file_type
```

Comment:
Should we maybe extract the has to a dedicated variable, e.g., `default_visibility` to reduce the file length (and make it easier to understand)?

Reply:
Do you mean line length of file length?

Reply:
I was referring to the line length. Hence, extracting it (as you did) is great -- thanks!